\documentclass[10pt,oneside]{article}

%%%%%%%%%%%%%
\setlength{\textheight}{8.75in} %Letter is 11in, less 2 for margins, less 0.25 for footer
\setlength{\oddsidemargin}{0.0in} %gets +1inc
\setlength{\evensidemargin}{0.0in} %gets +1inch
\setlength{\textwidth}{6.50in} %Letter is 8.5, less 2 inches for margins
\setlength{\topmargin}{0.5in}
\setlength{\headheight}{0in}
\setlength{\headsep}{0in}
\setlength{\parindent}{0.25in}
%%%%%%%%%%%%

% use letters instead of symbols to accomodate >7 authors
\makeatletter
\let\@fnsymbol\@alph
\makeatother

\usepackage[utf8]{inputenc}
\usepackage[numbers]{natbib}
\usepackage{graphicx}
\usepackage[colorlinks=true,citecolor=black,urlcolor=blue]{hyperref}

\title{%Hack to get the logo on the PDF front page:
\vspace{-1.5in}
\includegraphics[width=0.4\textwidth]{biopython.jpg}\\
\vspace{3mm}Biopython Project Update 2017}
\author{
    % Author sorting: alphabetically, except the first author.
	\underline{Sourav Singh}\thanks{Affiliation. Email: \href{mailto:ssouravsingh12@gmail.com}{ssouravsingh12@gmail.com}},
    Tiago Ant\~{a}o\thanks{Division of Biological Sciences, University of Montana, Montana, USA},
    Wibowo Arindrarto\thanks{Sequencing Analysis Support Core, Leiden University Medical Center, Leiden, NL},
    Christian Brueffer\thanks{Department of Clinical Sciences, Lund University, Lund, SE},
    Peter Cock\thanks{Information and Computational Sciences, James Hutton Institute (formerly SCRI), Invergowrie, Dundee, UK},\\
    Michiel de Hoon\thanks{Division of Genomic Technologies, RIKEN Center for Life Science Technologies, Yokohama, JP},
    Markus Piotrowski\thanks{Department of Plant Physiology, Ruhr-Universit√§t Bochum, DE},
    Leighton Pritchard\textsuperscript{e},  % same as Peter, saves space
    Connor T. Skennerton\thanks{Division of Geological and Planetary Sciences, California Institute of Technology, Pasadena, CA, USA},\\
    Eric Talevich\thanks{Department of Dermatology, University of California San Francisco, San Francisco, CA, USA},
    and the Biopython Contributors}
\date{18\textsuperscript{th} Bioinformatics Open Source Conference (BOSC) 2017, Prague, CZ}

\begin{document}
\maketitle
\thispagestyle{empty}

\vspace{-0.2in}
\noindent
Website: \url{http://biopython.org} \\
Repository: \url{https://github.com/biopython/biopython} \\
License: Biopython License Agreement (MIT style, see \url{http://www.biopython.org/DIST/LICENSE}); transition to 3-Clause BSD License ongoing\\

The Biopython Project is a long-running distributed collaborative effort,
supported by the Open Bioinformatics Foundation, which develops a freely
available Python library for biological computation \cite{AppNote}.

We present here details of the latest Biopython release - version 1.69.  With this
release we started to dual-license Biopython under both our original liberal
"Biopython License Agreement", and the very similar but more commonly used
"3-Clause BSD License".  In this release a small number of the Python files
are explicitly available under either license, but most of the code remains
under the "Biopython License Agreement" only.  New features in this release
include:

We are currently preparing a new release -- version 1.70 -- that includes$\cdots$

Additionally, a number of small bugs have been fixed with further additions
to the test suite, and there has been further work to follow the Python PEP8
and best practice standard coding style.

Our continuous integration process on GitHub has been enhanced by
using Tox to check for PEP8 style violations.


\begin{thebibliography}{}

\bibitem[Cock {\it et al}., 2009]{AppNote}Cock, P.J.A., Antao, T., Chang, J.T., Chapman, B.A., Cox, C.J., Dalke, A., Friedberg, I., Hamelryck, T., Kauff, F., Wilczynski, B., de Hoon, M.J. (2009) Biopython: freely available Python tools for computational molecular biology and bioinformatics. {\it Bioinformatics} {\bf 25}(11) 1422-3. \href{http://dx.doi.org/10.1093/bioinformatics/btp163}{doi:10.1093/bioinformatics/btp163}

\end{thebibliography}

\end{document}
