\documentclass[10pt,oneside]{article}

%%%%%%%%%%%%%
\setlength{\textheight}{8.75in} %Letter is 11in, less 2 for margins, less 0.25 for footer
\setlength{\oddsidemargin}{0.0in} %gets +1inc
\setlength{\evensidemargin}{0.0in} %gets +1inch
\setlength{\textwidth}{6.50in} %Letter is 8.5, less 2 inches for margins
\setlength{\topmargin}{0.5in}
\setlength{\headheight}{0in}
\setlength{\headsep}{0in}
\setlength{\parindent}{0.25in}
%%%%%%%%%%%%

% use letters instead of symbols to accommodate >7 authors
\makeatletter
\let\@fnsymbol\@alph
\makeatother

\usepackage[utf8]{inputenc}
\usepackage[numbers]{natbib}
\usepackage{graphicx}
\usepackage[colorlinks=true,citecolor=black,urlcolor=blue]{hyperref}

\title{%Hack to get the logo on the PDF front page:
\vspace{-1.5in}
\includegraphics[width=0.4\textwidth]{../presentation/figures/biopython.jpg}\\
\vspace{3mm}Biopython Project Update 2017}
\author{
	\underline{Sourav Singh}\thanks{Dept. of Computer Engineering, VIIT, Pune, India. Email: \href{mailto:ssouravsingh12@gmail.com}{ssouravsingh12@gmail.com}},
    Christian Brueffer\thanks{Department of Clinical Sciences, Lund University, Lund, SE},
    Peter Cock\thanks{Information and Computational Sciences, James Hutton Institute, Invergowrie, Dundee, UK},\\
    and the Biopython Contributors\thanks{See \href{https://github.com/biopython/biopython/blob/master/CONTRIB.rst}{contributor listing on GitHub}.}}
\date{18\textsuperscript{th} Bioinformatics Open Source Conference (BOSC) 2017, Prague, CZ}

\begin{document}
\maketitle
\thispagestyle{empty}

\vspace{-0.2in}
\noindent
Website: \url{http://biopython.org} \\
Repository: \url{https://github.com/biopython/biopython} \\
License: Biopython License Agreement (BSD like, see \url{http://www.biopython.org/DIST/LICENSE}) \\

The Biopython Project is a long-running distributed collaborative effort,
supported by the Open Bioinformatics Foundation, which develops a freely
available Python library for biological computation \cite{AppNote}.

We present here details of the latest Biopython release -- version 1.69.
This represents eight months of contributions, and a record 49 named
contributors including 28 newcomers which reflects our policy of trying to
encourage even small contributions.

Biopython 1.69 represents the start of our re-licensing plan, to transition away
from our liberal but unique \emph{Biopython License Agreement} to the similar
but very widely used \emph{3-Clause BSD License}. We are reviewing the code
base authorship file-by-file, in order to gradually dual license the entire
project.

New features include: a new parser for the ExPASy Cellosaurus cell line
database, catalogue and ontology; Bio.AlignIO now supports the UCSC Multiple
Alignment Format (MAF), including indexed access to large files using SQLite3;
Bio.SearchIO.AbiIO can now parse FSA files; an extended Bio.Affy module supporting
version 4 of the Affymetrix CEL format; updated Uniprot parsers to support
the ``submittedName'' XML element and features with unknown locations; better
handling of internal node comments in the NEXUS parser to improve compatibility
with tools such as BEAST TreeAnnotator; an update to Bio.Restriction to include
the REBASE February 2017 restriction enzyme list; updated Bio.SeqIO parsers for
GenBank, EMBL, and IMGT that now record the molecule type from the LOCUS/ID line
in the record.annotations dictionary and can cope with more format variations;
Bio.PDB.PDBList now can download PDBx/mmCif (new default), PDB (old default),
PDBML/XML and mmtf format protein structures; enhanced PyPy support by taking
advantage of NumPy and compiling most of the Biopython C code modules; the Bio.Seq
module now offers a complement function for consistency and a SeqFeature object's
qualifiers attribute is now an explicitly ordered dictionary.

Additionally we fixed miscellaneous bugs, enhanced our test suite and continued our
efforts to abide by the PEP8 and PEP257 coding style guidelines which is now checked
automatically with GitHub-integrated continous integration testing using TravisCI.
Current efforts include improving the unit test coverage, which is easily viewed
online at \href{https://codecov.io/github/biopython/biopython/}{CodeCov.io}.

We are currently preparing a new release -- version 1.70 -- that will feature an
extended Bio.AlignIO module that supports Mauve's eXtended Multi-FastA (XMFA) file format.

\begin{thebibliography}{}

\bibitem[Cock {\it et al}., 2009]{AppNote}Cock, P.J.A., Antao, T., Chang, J.T., Chapman, B.A., Cox, C.J., Dalke, A., Friedberg, I., Hamelryck, T., Kauff, F., Wilczynski, B., de Hoon, M.J. (2009) Biopython: freely available Python tools for computational molecular biology and bioinformatics. {\it Bioinformatics} {\bf 25}(11) 1422-3. \href{http://dx.doi.org/10.1093/bioinformatics/btp163}{doi:10.1093/bioinformatics/btp163}

\end{thebibliography}

\end{document}
