%\documentclass[trans,handout]{beamer}
\documentclass[trans]{beamer}
\usetheme{eit}

%\usepackage{pgfpages}
%\pgfpagesuselayout{4 on 1}[a4paper,landscape,border shrink=5mm]
%\pgfpagesuselayout{2 on 1}[a4paper,border shrink=5mm]

\usepackage[utf8]{inputenc}
\usepackage{graphicx}
\usepackage{textpos}
\usepackage{textcomp}
\usepackage{floatrow}
\usepackage{hyperref}
\hypersetup{
  colorlinks=true,
  linkcolor=black,
  urlcolor=blue
}
\graphicspath{{figures/}}

\title{
  \includegraphics[height=.2\textheight]{../abstract/biopython.jpg}\\[1em]
  Biopython Project Update 2016}
\subtitle{}
\author[Christian Brueffer]{
  \textbf{Christian Brueffer}*, Tiago Ant\~{a}o, Peter Cock, Eric Talevich,\\
  Michiel de Hoon, Wibowo Arindrarto, Leighton Pritchard,\\
  Anuj Sharma, Eric Rasche, Aaron Rosenfeld, Connor T.\\
  Skennerton, Marco Galardini, Markus Piotrowski,\\
  and the Biopython Contributors}
\institute[Translational Oncogenomics Unit, Department of Clinical Sciences, Lund University]{* Twitter \& GitHub: @cbrueffer\\Translational Oncogenomics Unit\\Department of Clinical Sciences \\
  Lund University\\
  Sweden\\[1em]
  Bioinformatics Open Source Conference 2016, Orlando, USA \\[1em]
  %\includegraphics[height=.2\textheight]{../abstract/biopython.jpg}
}
\date{July 9th, 2016}


\setcounter{tocdepth}{2}
\setbeamertemplate{caption}{\insertcaption}


% ToC at the beginning of every section
%\AtBeginSection[]
%{
 % \begin{frame} % with <beamer> => doesn't appear in handout mode
  %  \frametitle{Outline} %% Put the title you want, or none!
  %  \tableofcontents[currentsection,currentsection]
 % \end{frame}
%}

\begin{document}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}
	\titlepage
\end{frame}
\setbeamertemplate{footline}[body]

\begin{frame}
    \frametitle{Outline}
    \tableofcontents
\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{The Biopython Project}
\frame
{
  \frametitle{What is Biopython?}

  \begin{itemize}
  \item Collection of modules for biological computation in Python
  \begin{itemize}
  \item Sequence handling and motifs, parsers, database queries, protein structures, phylogenetics, tool wrappers and lots more
  \end{itemize}
  \item Started in 1999, first release in 2000
  \item Open source and freely available (Biopython license)
  \end{itemize}

  \begin{center}
  \includegraphics[width=0.9\textwidth]{openhub-bp-nutshell.png}\\
  \tiny{Source: \url{https://www.openhub.net/p/biopython}}
  \end{center}
}
\frame
{
  \frametitle{Stats for the last 12 months}

  %\begin{center}
  %\includegraphics[width=0.9\textwidth]{openhub-bp-activity.png}
  %\end{center}

  \begin{center}
  %\includegraphics[width=0.9\textwidth]{openhub-bp-community.png}
  \includegraphics[width=1.0\textwidth]{openhub-bp-activity-community.png}
  \end{center}
  \small{Source: \url{https://www.openhub.net/p/biopython}}
}
\frame
{
  \frametitle{Lots of new contributors!}
% 1.66
  Alan Medlar, Anthony Mathelier, Antony Lee, Anuj Sharma, Ben Fulton, Bertrand Néron, Connor T. Skennerton, David Arenillas, David Nicholson, Emmanuel Noutahi, Eric Rasche, Fabio Madeira, Franco Caramia, Gert Hulselmans, Gleb Kuznetsov, John Bradley, Kian Ho, Kozo Nishida, Kuan-Yi Li, Sunhwan Jo, Tarcisio Fedrizzi,

%1.67
  Aaron Rosenfeld, Anders Pitman, Barbara Mühlemann, Ben Woodcroft, Brian Osborne, Chaitanya Gupta, Chris Warth, Christiam Camacho, David Koppstein, Jacek Śmietański, João D Ferreira, Joe Cora, Marco Galardini, Matt Ruffalo, Matteo Sticco, Nader Morshed, Owen Solberg, Steve Bond, Terry Jones,

% after
  Anthony Bradle, Uwe Schmitt
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{New Releases and Beyond}
\subsection*{Release 1.66}
\frame
{
  \frametitle{Biopython 1.66 (released 2015-10-21)}

  \begin{itemize}
  \item extended Bio.KEGG and Bio.Graphics modules to support drawing KEGG pathways with transparency
  \item extended ``abi'' Bio.SeqIO parser to decode almost all documented fields used by ABIF instruments
  \item QCPSuperimposer module using the Quaternion Characteristic Polynomial algorithm for superimposing structures to Bio.PDB
  \item extended Bio.Entrez module to implement the NCBI Entrez Citation Matching function and to support NCBI XML files with XSD schemas.
  \item miscellaneous bug fixes, test suite enhancement, better PEP8 coding style adherence
  \item Python 2.6 support deprecated
  \end{itemize}
}
\subsection*{Release 1.67}
\frame
{
  \frametitle{Biopython 1.67 (released 2016-06-08)}

  \begin{itemize}
  \item deprecated ``=='' SeqRecord comparison
  \item Bio.phenotype module for working with Phenotype Microarray data
  \item updates to Bio.Data to include NCBI genetic code table 25 (covering Candidate Division SR1 and Gracilibacteria)
  \item update to Bio.Restriction to include the REBASE May 2016 restriction enzyme list
  \item updates to BioSQL to use foreign keys with SQLite3 databases
  \item corrections to the Bio.Entrez module and the MMCIF structure parser
  \item Python 3.3 support deprecated
  \end{itemize}
}
\subsection*{Current Developments}
\frame
{
  \frametitle{Current Developments}

  \begin{itemize}
  \item Bio.PDB extended to parse the RSSB's new binary Macromolecular
Transmission Format (MMTF)

  \item Module Bio.pairwise2 has been re-written (faster, addresses some problems with local alignments, and also now allows gap insertions after deletions, and vice versa)

  \item The sample graphical tool SeqGui (Sequence Graphical User Interface) was
rewritten using the tkinter library (contributed by Markus Piotrowski). This
allows simple nucleotide transcription, back-transcription and translation
into amino acids using Bio.Seq internally, offering of the NCBI genetic codes
supported in Biopython.

  \item New NCBI genetic code table 26 (Pachysolen tannophilus Nuclear Code) has been
added to Bio.Data (and the translation functionality)

  \item In line with NCBI website changes, Biopython now uses HTTPS rather than HTTP
to connect to the NCBI Entrez and QBLAST API.

  \item Bug fixes, further additions to the test suite, more Python PEP8 style adherence

  \end{itemize}
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Docker Images}
\frame
{
  \frametitle{Docker Images}

  \begin{itemize}
  \item Previously existing containers:
    \begin{itemize}
    \item container with Python 2/3, Biopython and all dependencies
    \item BioSQL container
  \end{itemize}
  \item New: Jupyter notebook containers
    \begin{itemize}
      \item basic version
      \item version including Biopython tutorial as notebooks
    \end{itemize}
  \end{itemize}
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{General Updates}
\frame
{
  \frametitle{Continuous Integration}

  \begin{itemize}
  \item TravisCI
  \item Experimentation of different services
  \begin{itemize}
  \item Codecov.io (test coverage)
  \item Quantified Code (automatic pull requests with Cody)
  \item Landscape.io (``health score'')
  \end{itemize}
  \item Currently enabled by default: CodeCov.io
  \end{itemize}

  \begin{columns}
  \column{0.4\textwidth}
  \includegraphics[width=1\textwidth]{bp-codecov.png}
  \column{0.6\textwidth}
  \includegraphics[width=1\textwidth]{bp-landscape.png}
  \end{columns}
}

\frame
{
  \frametitle{Website}

  \begin{itemize}
  \item Previously hosted by OBF until server problems
  \item Migrated from OBF MediaWiki to GitHub Pages
  \item Repository: \url{https://github.com/biopython/biopython.github.io}
  \end{itemize}

  \begin{center}
  \includegraphics[width=0.9\textwidth]{bp-website.png}
  \end{center}
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Conclusion}
\frame
{
  \frametitle{Conclusion}

  \begin{itemize}
  \item Lots of new stuff
  \item Lots of new contributors
  \end{itemize}
}

\section*{Acknowledgements}
\frame
{
  \frametitle{Resources!}

  %\begin{center}
  Website:\\
  \begin{itemize}
  \item \url{http://biopython.org}
  \end{itemize}

  Repositories:\\
  \begin{itemize}
  \item Main: \url{http://github.com/biopython/biopython}
  \item Docker: \url{http://github.com/biopython/biopython_docker}
  \item Website: \url{https://github.com/biopython/biopython.github.io}
  \end{itemize}

  Mailing lists:
  \begin{itemize}
  \item General list: \url{biopython@biopython.org}
  \item Developers list: \url{biopython-dev@biopython.org}
  \end{itemize}

  Biostars:
  \begin{itemize}
  \item \url{https://www.biostars.org/t/biopython/} (``biopython'' category)
  \end{itemize}
  %\end{center}
}

\frame
{
  \frametitle{Thanks to...}

  Biopython

  \begin{itemize}
  \item Peter Cock
  \item Biopython Community
  \end{itemize}

  Lund University

  \begin{itemize}
  \item Lao Saal (PhD supervisor)
  \item Faculty of Medicine (Travel Grant)
  \end{itemize}
}

\end{document}