%\documentclass[trans,handout]{beamer}
\documentclass[trans]{beamer}
\usetheme{eit}

%\usepackage{pgfpages}
%\pgfpagesuselayout{4 on 1}[a4paper,landscape,border shrink=5mm]
%\pgfpagesuselayout{2 on 1}[a4paper,border shrink=5mm]

\usepackage[utf8]{inputenc}
\usepackage{graphicx}
\usepackage{textpos}
\usepackage{textcomp}
\usepackage{floatrow}
\usepackage{multicol}
\usepackage{hyperref}
\hypersetup{
  colorlinks=true,
  linkcolor=black,
  urlcolor=blue
}
\graphicspath{{figures/}}

\title{
  \includegraphics[height=.2\textheight]{../abstract/biopython.jpg}\\[1em]
  Biopython Project Update 2016}
\subtitle{}
\author[Christian Brueffer]{
  \textbf{Christian Brueffer}*, Tiago Ant\~{a}o, Peter Cock, Eric Talevich,\\
  Michiel de Hoon, Wibowo Arindrarto, Leighton Pritchard,\\
  Anuj Sharma, Eric Rasche, Aaron Rosenfeld, Connor T.\\
  Skennerton, Marco Galardini, Markus Piotrowski,\\
  and the Biopython Contributors}
\institute[Translational Oncogenomics Unit, Department of Clinical Sciences, Lund University]{* Twitter \& GitHub: @cbrueffer\\Translational Oncogenomics Unit\\Department of Clinical Sciences \\
  Lund University\\
  Sweden\\[1em]
  Bioinformatics Open Source Conference 2016, Orlando, USA \\[1em]
  %\includegraphics[height=.2\textheight]{../abstract/biopython.jpg}
}
\date{July 9th, 2016}


\setcounter{tocdepth}{2}
\setbeamertemplate{caption}{\insertcaption}


% ToC at the beginning of every section
%\AtBeginSection[]
%{
 % \begin{frame} % with <beamer> => doesn't appear in handout mode
  %  \frametitle{Outline} %% Put the title you want, or none!
  %  \tableofcontents[currentsection,currentsection]
 % \end{frame}
%}

\begin{document}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{frame}
	\titlepage
\end{frame}
\setbeamertemplate{footline}[body]

\begin{frame}
    \frametitle{Outline}
    \tableofcontents
\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{The Biopython Project}
\frame
{
  \frametitle{What is Biopython?}

  \begin{itemize}
  \item Collection of modules for biological computation in Python
  \begin{itemize}
  \item Sequence handling and motifs, parsers, database queries, protein structures, phylogenetics, tool wrappers and lots more
  \end{itemize}
  \item Started in 1999, first release in 2000
  \item Open source and freely available (Biopython license)
  \end{itemize}

  \begin{center}
  \includegraphics[width=0.9\textwidth]{openhub-bp-nutshell.png}
  \end{center}
  \small{Source: \url{https://www.openhub.net/p/biopython}}
}
\frame
{
  \frametitle{Stats for the previous 12 months}

  \begin{center}
  \includegraphics[width=1.0\textwidth]{openhub-bp-activity-community.png}
  \end{center}
  \small{Source: \url{https://www.openhub.net/p/biopython}}
}
\frame
{
  \frametitle{Lots of new contributors!}
  
  \scriptsize{
  \begin{multicols}{3}
  \begin{itemize}
  % 1.66
  \item Alan Medlar
  \item Anthony Mathelier
  \item Antony Lee
  \item Anuj Sharma
  \item Ben Fulton
  \item Bertrand Néron
  \item Connor T. Skennerton
  \item David Arenillas
  \item David Nicholson
  \item Emmanuel Noutahi
  \item Eric Rasche
  \item Fabio Madeira
  \item Franco Caramia
  \item Gert Hulselmans
  \item Gleb Kuznetsov
  \item John Bradley
  \item Kian Ho
  \item Kozo Nishida
  \item Kuan-Yi Li
  \item Sunhwan Jo
  \item Tarcisio Fedrizzi

  %1.67
  \item Aaron Rosenfeld
  \item Anders Pitman
  \item Barbara Mühlemann
  \item Ben Woodcroft
  \item Brian Osborne
  \item Chaitanya Gupta
  \item Chris Warth
  \item Christiam Camacho
  \item David Koppstein
  \item Jacek Śmietański
  \item João D Ferreira
  \item Joe Cora
  \item Marco Galardini
  \item Matt Ruffalo
  \item Matteo Sticco
  \item Nader Morshed
  \item Owen Solberg
  \item Steve Bond
  \item Terry Jones

  % after
  \item Anthony Bradle
  \item Uwe Schmitt
  \end{itemize}
  \end{multicols}
  }
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{New Releases and Beyond}
\subsection*{Release 1.66}
\frame
{
  \frametitle{Biopython 1.66 (released 2015-10-21)}

  \begin{itemize}
  \item extended Bio.KEGG and Bio.Graphics modules to support drawing KEGG pathways with transparency (Leighton Pritchard)
  \item extended ``abi'' Bio.SeqIO parser to decode almost all documented fields used by ABIF instruments (Eric Rasche)
  \item QCPSuperimposer module using the Quaternion Characteristic Polynomial algorithm for superimposing structures to Bio.PDB (Anuj Sharma)
  \item extended Bio.Entrez module to implement the NCBI Entrez Citation Matching function and to support NCBI XML files with XSD schemas (Aaron Rosenfeld)
  \item miscellaneous bug fixes, test suite enhancement, better PEP8 coding style adherence
  \item Python 2.6 support deprecated
  \end{itemize}
}
\frame
{
  \frametitle{Biopython 1.66 Contributors}

  \scriptsize{
  \begin{multicols}{2}
  \begin{itemize}
  \item Alan Medlar (*)
  \item Anthony Mathelier (*)
  \item Antony Lee (*)
  \item Anuj Sharma (*)
  \item Ben Fulton (*)
  \item Bertrand Néron (*)
  \item Brandon Invergo
  \item Carlos Pena
  \item Christian Brueffer
  \item Connor T. Skennerton (*)
  \item David Arenillas (*)
  \item David Nicholson (*)
  \item Emmanuel Noutahi (*)
  \item Eric Rasche (*)
  \item Fabio Madeira (*)
  \item Franco Caramia (*)
  \item Gert Hulselmans (*)
  \item Gleb Kuznetsov (*)
  \item João Rodrigues
  \item John Bradley (*)
  \item Kai Blin
  \item Kian Ho (*)
  \item Kozo Nishida (*)
  \item Kuan-Yi Li (*)
  \item Leighton Pritchard
  \item Lucas Sinclair
  \item Michiel de Hoon
  \item Peter Cock
  \item Saket Choudhary
  \item Sunhwan Jo (*)
  \item Tarcisio Fedrizzi (*)
  \item Tiago Antao
  \item Vincent Davis
  \end{itemize}
  \end{multicols}
  }
}

\subsection*{Release 1.67}
\frame
{
  \frametitle{Biopython 1.67 (released 2016-06-08)}

  \begin{itemize}
  \item deprecated ``=='' SeqRecord comparison (David Koppstein)
  \item Bio.phenotype module for working with Phenotype Microarray data (Marco Galardini)
  \item updates to Bio.Data to include NCBI genetic code table 25 (covering Candidate Division SR1 and Gracilibacteria)
  \item update to Bio.Restriction to include the REBASE May 2016 restriction enzyme list (Markus Piotrowski)
  \item updates to BioSQL to use foreign keys with SQLite3 databases (Connor T. Skennerton)
  \item corrections to the Bio.Entrez module (Aaron Rosenfeld)
  \item corrections to the MMCIF structure parser
  \item Python 3.3 support deprecated
  \end{itemize}
}
\frame
{
  \frametitle{Biopython 1.67 Contributors}

  \scriptsize{
  \begin{multicols}{2}
  \begin{itemize}
  \item Aaron Rosenfeld (*)
  \item Anders Pitman (*)
  \item Barbara Mühlemann (*)
  \item Ben Fulton
  \item Ben Woodcroft (*)
  \item Brandon Invergo
  \item Brian Osborne (*)
  \item Carlos Pena
  \item Chaitanya Gupta (*)
  \item Chris Warth (*)
  \item Christiam Camacho (*)
  \item Connor T. Skennerton
  \item David Koppstein (*)
  \item Eric Talevich
  \item Jacek Śmietański (*)
  \item João D Ferreira (*)
  \item João Rodrigues
  \item Joe Cora (*)
  \item Kai Blin
  \item Leighton Pritchard
  \item Lenna Peterson
  \item Marco Galardini (*)
  \item Markus Piotrowski
  \item Matt Ruffalo (*)
  \item Matteo Sticco (*)
  \item Nader Morshed (*)
  \item Owen Solberg (*)
  \item Peter Cock
  \item Steve Bond (*)
  \item Terry Jones (*)
  \item Vincent Davis
  \item Zheng Ruan
  \end{itemize}
  \end{multicols}
  }
}

\subsection*{Current Developments}
\frame
{
  \frametitle{Current Developments}

  \begin{itemize}
  \item Bio.PDB extended to parse the RSSB's new binary Macromolecular
Transmission Format (MMTF) Anthony Bradle -> Poster 22

  \item Module Bio.pairwise2 has been re-written (faster, addresses some problems with local alignments, and also now allows gap insertions after deletions, and vice versa) -> Markus Piotrowski

  \item The sample graphical tool SeqGui (Sequence Graphical User Interface) was
rewritten using the tkinter library (contributed by Markus Piotrowski). This
allows simple nucleotide transcription, back-transcription and translation
into amino acids using Bio.Seq internally, offering of the NCBI genetic codes
supported in Biopython. -> Markus Piotrowski

  \item New NCBI genetic code table 26 (Pachysolen tannophilus Nuclear Code) has been
added to Bio.Data (and the translation functionality)

  \item In line with NCBI website changes, Biopython now uses HTTPS rather than HTTP
to connect to the NCBI Entrez and QBLAST API. (Peter Cock)

  \item Bug fixes, further additions to the test suite, more Python PEP8 style adherence

  \end{itemize}
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Docker Images}
\frame
{
  \frametitle{Docker Images}

  \begin{itemize}
  \item Previously existing containers:
    \begin{itemize}
    \item container with Python 2/3, Biopython and all dependencies
    \item BioSQL container
  \end{itemize}
  \item New: Jupyter notebook containers
    \begin{itemize}
      \item basic version
      \item version including Biopython tutorial as notebooks
    \end{itemize}
  \end{itemize}
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{General Updates}
\frame
{
  \frametitle{Continuous Integration}

  \begin{itemize}
  \item TravisCI
  \item Experimentation of different services
  \begin{itemize}
  \item Codecov.io (test coverage)
  \item Quantified Code (automatic pull requests with Cody)
  \item Landscape.io (``health score'')
  \end{itemize}
  \item Currently enabled by default: CodeCov.io
  \end{itemize}

  \begin{columns}
  \column{0.4\textwidth}
  \includegraphics[width=1\textwidth]{bp-codecov.png}
  \column{0.6\textwidth}
  \includegraphics[width=1\textwidth]{bp-landscape.png}
  \end{columns}
}

\frame
{
  \frametitle{Website}

  \begin{itemize}
  \item Previously hosted by OBF until server problems
  \item Migrated from OBF MediaWiki to GitHub Pages
  \item Repository: \url{https://github.com/biopython/biopython.github.io}
  \end{itemize}

  \begin{center}
  \includegraphics[width=0.9\textwidth]{bp-website.png}
  \end{center}
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Conclusion}
\frame
{
  \frametitle{Conclusion}

  \begin{itemize}
  \item Lots of new stuff
  \item Lots of new contributors
  \end{itemize}
}

\section*{Acknowledgements}
\frame
{
  \frametitle{Resources!}

  %\begin{center}
  Website:\\
  \begin{itemize}
  \item \url{http://biopython.org}
  \end{itemize}

  Repositories:\\
  \begin{itemize}
  \item Main: \url{http://github.com/biopython/biopython}
  \item Docker: \url{http://github.com/biopython/biopython_docker}
  \item Website: \url{https://github.com/biopython/biopython.github.io}
  \end{itemize}

  Mailing lists:
  \begin{itemize}
  \item General list: \url{biopython@biopython.org}
  \item Developers list: \url{biopython-dev@biopython.org}
  \end{itemize}

  Biostars:
  \begin{itemize}
  \item \url{https://www.biostars.org/t/biopython/} (``biopython'' category)
  \end{itemize}
  %\end{center}
}

\frame
{
  \frametitle{Acknowledgements}

  \begin{minipage}{1\textwidth}
  \begin{columns}
  \column{0.5\textwidth}
  \begin{itemize}
  \item Peter Cock
  \item Biopython Community
  \end{itemize}
  \column{0.5\textwidth}
  \includegraphics[width=0.8\textwidth]{../abstract/biopython.jpg}
  \end{columns}
  \end{minipage}

  \vspace{0.5cm}

  \begin{minipage}{1\textwidth}
  \begin{columns}
  \column{0.5\textwidth}
  \begin{itemize}
  \item Lao Saal (PhD supervisor)
  \item Faculty of Medicine
  \end{itemize}
  \column{0.5\textwidth}
  \includegraphics[width=0.5\textwidth]{LundUniversity_C2line_PMS.eps}
  \end{columns}
  \end{minipage}
  
  \vspace{0.5cm}  
  
  \begin{minipage}{1\textwidth}
  \begin{columns}
  \column{0.25\textwidth}
  \includegraphics[width=0.5\textwidth]{github-logo.jpg}
  \column{0.25\textwidth}
  \includegraphics[width=0.5\textwidth]{travisci-logo.png}
  \column{0.25\textwidth}
  \includegraphics[width=0.5\textwidth]{codecov-logo.png}
  \column{0.25\textwidth}
  \includegraphics[width=0.5\textwidth]{quantifiedcode-logo.png}
  \end{columns}
  \end{minipage}
}

\end{document}